### ISSUE TITLE
Persistent AudioSync Delay on SteamDeck

### ISSUE THREAD

**Issue by Gildo8o** [streetpea/chiaki4deck#137]
Chiaki4deck works great up until a lag spike which causes audio to be delayed by a few seconds and persists until the session ends or you restart the remote play connection. Tried different settings to try and mitigate issue with no luck.

**Comment by darkdirtydwarf** [comment]
Same here. At the beginning of the session I have very little delay (probably as good as technically possible) but the more I play, the greater the delay becomes.

I have an OLED SD. I tried changing all possible options but didn’t find a solution. Some options seem to make things better (seeing the audio output to default instead of auto) but then I ran into another problem (chiaki4deck freezes after about 20 minutes playing.

I just read someone had better luck removing port forwarding, I’m gonna try that next (not sure why it would help though).

**Comment by streetpea** [comment]
Have you tried decreasing the audio buffer?

**Comment by darkdirtydwarf** [comment]
In my case it might be related to port forwarding (even if I don't understand how, got no expertise in this field).

Not a definitive answer, as I need to test again to confirm in the next days, but as someone else reported in another issues thread (sorry, I can't find it right now), removing the port forwarding for the PS5 in the router settings seems to have resolved the issue: over 40 minutes gameplay, no freeze (usually happening before 30 min), no sound delay (before it was starting increasing starting at about 5 to 10 minutes into a session) and actually very smooth experience, I couldn't really tell I was playing remotely.
I kept pausing the game to check the delay in the PS5 menu, it didn't change during the session. Fingers crossed that it stays this way.

Settings were all set as recommended for PS5 HDR on SD OLED; HDMI cable NOT disconnected but HDMI CEC disabled in the PS5 menu (some people claim this plays a role too).

**Comment by darkdirtydwarf** [comment]
It probably has nothing to do with the port forwarding, at least in my case. It seems more related to the fact that the SD keeps choosing the wrong WiFi with worse signal.

I've been playing again today. I was having the problem again with no port forwarding set on the router (several seconds of delay and increasing during the session). I noticed the SD was connected to a repeater one floor down, although at the beginning of the session I had set the connection to the router next to me. I forced the connection to the router by deleting every other WiFi and it then worked flawlessly for over an hour. 

Given all the issues I have with the SD WiFi settings, I assume the problem isn't with chiaki4deck. 

Odd enough, when I was having the delay, I rebooted the SD, connected to the PS5 again (which I didn't reboot but left on) and still had the same delay (before it would "reset" when relaunching chiaky4deck).
